### DISCUSSION TITLE
Migrating from NSPanel-lovelace-ui

### DISCUSSION THREAD

**Discussion by mcrodriguezb** [Blackymas/NSPanel_HA_Blueprint#2138] [Q&A] [answered]
Hi all,

I've decided to move to the blueprint approach and need to update my 5 NSPanel. I read the documentation and also found[ issue 223](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/223). I know issue is from Jan 2023, but I have some questions about the migration. 

As stated in 223 by @pawellygan and it is align with the documentation, these should be the steps:
1. Download the file [nspanel_blank.tft](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/hmi/nspanel_blank.tft) from our GitHub repository.
2. Upload the same file (nspanel_blank.tft) to the folder "www" in your Home Assistant
3. Double check if the file is available by trying to download it using your local url (typically http://homeassistant.local:8123/local/nspanel_blank.tft).
4. in the tasmota console execute (via FlashNextion http://homeassistant.local:8123/local/nspanel_blank.tft)
5. After executing a screen with the QR code should be presented in the nspanel.
6. Create in ESP Home UI a new device as explained in the documentation. 
7. Click install and select manual download. Once compilation is completed, ask to download as modern or legacy. Select legacy.
8. Firmware upgrade through tasmota UI with the previous generated firmware (step 7)

From here, all are assumptions:
10. blank again as a HA service (local file): ESP device identified and created as a HA device.
11. ESP Home update via ESP Home (dropping legacy mode): Install a new firmware using ESPHome and wireless to drop legacy mode.
12. From HA select to update the TFT in the nspanel device.

Please, anyone who has gone through this migration, can you confirm the steps?

I've also seen issue 166 Error: Model does not match-Nextion Display bricked-Solution. Is this solved? the solution mention about soldering to upload a new TFT.

Thanks a lot for the support and hard work!!!

Regards,
MC

PS: After doing the migration, I can do a PR to clarify the migration steps

**Comment by edwardtfn** [comment]
Sorry for the late reply here... I'm taking a long weekend to drive around some fjords in Norway. 😉    Again... 😂

Well, until step 8 all looks right. After that, the panel will automatically be found by your ESPHome Dashboard and will pop-up a new device on Home Assistant (where you only need to follow the steps in your screen).
After the device is discovered in HA, you only need to select your model and click to update TFT (step 12 from your list). No need to flash again with another firmware format.

About the issue 166, it can always happen when a TFT upload is incomplete, however the latest versions will recover that situation in 99.9% of the cases, so I wouldn't concern that much about that.

**Comment by mcrodriguezb** [comment]
Thanks for the clarification. I follow the steps and everything runs well in the migration and my nspanel is running with framework esp-idf. First firmware was installed using tasmota ui and legacy.

Then, I realized I had enable the heating and I want to disable it, but I face a problem in the new firmware installation. I found a similar issue in https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1791. I tried to clean the build files as suggested, but same error.

Any way to install the firmware using OTA?

**Comment by mcrodriguezb** [comment]
I continued investigating this issue and it could be related to the safeboot partition introduced by tasmota in version 12. It looks the only optin is to do a serial firmware installation.

I also found the following https://esphome.io/guides/migrate_sonoff_tasmota.html where it clearly say: "WARNING! Migrating via OTA on ESP32, from Tasmota v12 (with boot partition) and up is currently not possible! Trying it could soft-brick your device!"

@edwardtfn do you have any suggestion?

Thanks,
MC

**Comment by edwardtfn** [comment]
What do you have installed in your device right now? Tasmota or ESPHome?
Could you please share your logs with the error? I understood it is similar to #1791, but I wanna make sure we are not missing any detail.

You haven't tried to flash via serial, right?

**Reply by mcrodriguezb** [reply]
> What do you have installed in your device right now? Tasmota or ESPHome? 

Currently I have ESPHome.

> Could you please share your logs with the error? I understood it is similar to #1791, but I wanna make sure we are not missing any detail.

Same logs that in 1791.

> You haven't tried to flash via serial, right?

Not yet. I just unplugged the screen and almost open it. Unless you tell me another option, that's my next step.

**Reply by edwardtfn** [reply]
I don't have anything on mind. Serial flashing will probably solve all those issues.

**Reply by mcrodriguezb** [reply]
I just flashed and it worked. Before putting again in the wall, I checked a wireless installation and it worked.

@edwardtfn What would be the process for moving from nspanel-lovelace-ui to nspanel-ha-blueprint? Do we still need to do the following steps before doing the serial flashing?

1. Download the file [nspanel_blank.tft](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/hmi/nspanel_blank.tft) from our GitHub repository.
2. Upload the same file (nspanel_blank.tft) to the folder "www" in your Home Assistant
3. Double check if the file is available by trying to download it using your local url (typically http://homeassistant.local:8123/local/nspanel_blank.tft).
4. in the tasmota console execute (via FlashNextion http://homeassistant.local:8123/local/nspanel_blank.tft)

Thans for your support.

Regards,
MC

**Comment by mcrodriguezb** [comment, marked as answer]
For future reference. 

The first firmware works from tasmota UI, but as soon as you add any extension a serial flash is needed
